src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -1574,10 +1640,15 @@ mgp_error mgp_result_record_insert(mgp_result_record *record, const char *field_
     if (find_it == record->signature->end()) {
       throw std::out_of_range{fmt::format("The result doesn't have any field named '{}'.", field_name)};
     }
+    if (record->ignore_deleted_values && ContainsDeleted(val)) [[unlikely]] {
+      record->values.emplace(field_name, ToTypedValue(*NewRawMgpObject<mgp_value>(memory), memory));
```

Comment:
Do you need to emplace anything here if it has deleted values?

Reply:
Good catch, emplacing isn’t necessary in case of a deleted value because the record will get skipped anyway